File: Calculator/PHP/Web/Laravel/src/app/Http/Controllers/CalculatorController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;

class CalculatorController extends Controller
{
    public function calculate(Request $request): JsonResponse
    {
        $data = $request->validate([
            'a' => 'required|numeric',
            'b' => 'required|numeric',
            'operator' => 'required|string|in:add,subtract,multiply,divide',
        ]);

        $a = (float) $data['a'];
        $b = (float) $data['b'];
        $operator = $data['operator'];

        if ($operator === 'divide' && $b === 0.0) {
            return response()->json(['error' => 'Division by zero is not allowed'], 400);
        }

        $result = match ($operator) {
            'add' => $a + $b,
            'subtract' => $a - $b,
            'multiply' => $a * $b,
            'divide' => $a / $b,
        };

        return response()->json(['result' => $result]);
    }
}

File: Calculator/Java/Web/SpringBoot/src/main/java/com/portfolio/calculator/CalculatorController.java
package com.portfolio.calculator;

import org.springframework.http.ResponseEntity;
import org.springframework.http.converter.HttpMessageNotReadableException;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;
import java.util.Set;

@RestController
@RequestMapping("/api/calculate")
public class CalculatorController {
    private static final Set<String> OPERATORS = Set.of("add", "subtract", "multiply", "divide");

    @PostMapping
    public ResponseEntity<?> calculate(@RequestBody CalculateRequest request) {
        if (request.getA() == null || request.getB() == null) {
            return ResponseEntity.badRequest().body(Map.of("error", "Invalid number input"));
        }
        String operator = request.getOperator();
        if (operator == null || !OPERATORS.contains(operator)) {
            return ResponseEntity.badRequest().body(Map.of("error", "Invalid operator"));
        }
        if ("divide".equals(operator) && request.getB() == 0) {
            return ResponseEntity.badRequest().body(Map.of("error", "Cannot divide by zero"));
        }
        double result = Calculator.calculate(request.getA(), request.getB(), operator);
        return ResponseEntity.ok(Map.of("result", result));
    }

    @ExceptionHandler(HttpMessageNotReadableException.class)
    public ResponseEntity<Map<String, String>> onUnreadable() {
        return ResponseEntity.badRequest().body(Map.of("error", "Invalid number input"));
    }
}
